internal/utils/bytes.go

```diff
@@ -89,7 +89,7 @@ func countOneBitsInByte(by byte) int {
 }
 
 // CountOneBits counts the number of 1 bit in byte array
-func CountOneBits(arr []byte) int {
+func CountOneBits(arr []byte) int64 {
```

Comment:
If we are changing this, I'd suggest using uint64 (a bit count can never be negative, can it?).